cairosvg/url.py

```diff
@@ -32,6 +32,64 @@
 URL = re.compile(r'url\((.+)\)')
 
 
+class URLFetcher(object):
```

Comment:
I'm not sure that we need a class with 5 methods. Giving a split API for overriden classes is nice when each method has a part of the logic in the default implementation, and you want other classes to keep some parts of this logic. But this default implementation has only 1 line that you can keep in other implementations.

I would only keep 1 function instead of the whole class.

```python
def fetch(self, url):
    return urlopen(Request(url, headers=HTTP_HEADERS)).read()
```

If some people want to create complex URL fetchers and keep code split into different logical functions, they can create classes with private methods and only the `__init__` method as an equivalent of the function call. Bonus: they will even be able to reuse the original `fetch` function if they want :wink:.

```python
from cairosvg.url import fetch

class URLFetcher:
    def __init__(self, url):
        self._validate(url)
        fetcher = self._fetcher_for(url)
        if fetcher:
            return fetcher(url)

    def _fetcher_for(self, url):
        if url.startswith('https://'):
            return self._fetch_https

    def _fetch_https(self, url):
        return fetch(url)

    def _validate(self, url):
        if 'forbidden' in url:
            raise ValueError
```

It's also possible to generate parametric URL fetchers using metaclasses, but, well…